src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -822,14 +823,20 @@ public ApiResponse handleApiView(String name, JSONObject params)
 			result = new ApiResponseElement(alertToSet(new Alert(recordAlert)));
 		} else if (VIEW_ALERTS.equals(name)) {
 			final ApiResponseList resultList = new ApiResponseList(name);
+			final int risk = this.getParam(params, PARAM_RISK, -1);
+			
 			processAlerts(
 					this.getParam(params, PARAM_BASE_URL, (String) null), 
 					this.getParam(params, PARAM_START, -1), 
 					this.getParam(params, PARAM_COUNT, -1), new Processor<Alert>() {
 
 						@Override
 						public void process(Alert alert) {
-							resultList.addItem(alertToSet(alert));
+							if (risk == -1) {
```

Comment:
This would be better if done in `processAlerts`, `numberOfAlerts` should also have the risk parameter.